### ISSUE TITLE
FnSplit for values that are generated from Intrinsic Functions fail as it's only expecting a strings.

### ISSUE THREAD

**Issue by cmaxwellau** [allinwonder/cfn2dsl#13]
cfn2dsl -t output/mx-database.json  ../mx-database-import.rb
/Users/camm/.rvm/gems/ruby-2.4.1/gems/awesome_print-1.7.0/lib/awesome_print/formatters/base_formatter.rb:31: warning: constant ::Fixnum is deprecated
/Users/camm/.rvm/gems/ruby-2.4.1/gems/cfn2dsl-0.3.1/lib/intrinsic_function.rb:120:in `+': no implicit conversion of IntrinsicFunction into String (TypeError)
	from /Users/camm/.rvm/gems/ruby-2.4.1/gems/cfn2dsl-0.3.1/lib/intrinsic_function.rb:120:in `fn_split'
	from /Users/camm/.rvm/gems/ruby-2.4.1/gems/cfn2dsl-0.3.1/lib/intrinsic_function.rb:11:in `to_s'
	from /Users/camm/.rvm/gems/ruby-2.4.1/gems/awesome_print-1.7.0/lib/awesome_print/formatter.rb:58:in `awesome_self'
	from /Users/camm/.rvm/gems/ruby-2.4.1/gems/awesome_print-1.7.0/lib/awesome_print/formatter.rb:36:in `format'
	from /Users/camm/.rvm/gems/ruby-2.4.1/gems/awesome_print-1.7.0/lib/awesome_print/inspector.rb:148:in `unnested'
	from /Users/camm/.rvm/gems/ruby-2.4.1/gems/awesome_print-1.7.0/lib/awesome_print/inspector.rb:115:in `awesome'
	from /Users/camm/.rvm/gems/ruby-2.4.1/gems/awesome_print-1.7.0/lib/awesome_print/core_ext/kernel.rb:10:in `ai'
	from (erubis):161:in `block (3 levels) in evaluate'
	from (erubis):157:in `each_pair'
	from (erubis):157:in `block (2 levels) in evaluate'
	from (erubis):82:in `each'
	from (erubis):82:in `block in evaluate'
	from /Users/camm/.rvm/gems/ruby-2.4.1/gems/erubis-2.7.0/lib/erubis/evaluator.rb:74:in `instance_eval'
	from /Users/camm/.rvm/gems/ruby-2.4.1/gems/erubis-2.7.0/lib/erubis/evaluator.rb:74:in `evaluate'
	from /Users/camm/.rvm/gems/ruby-2.4.1/gems/cfn2dsl-0.3.1/lib/render.rb:13:in `cfn_to_cfndsl'
	from /Users/camm/.rvm/gems/ruby-2.4.1/gems/cfn2dsl-0.3.1/bin/cfn2dsl:42:in `<top (required)>'
	from /Users/camm/.rvm/gems/ruby-2.4.1/bin/cfn2dsl:23:in `load'
	from /Users/camm/.rvm/gems/ruby-2.4.1/bin/cfn2dsl:23:in `<main>'
	from /Users/camm/.rvm/gems/ruby-2.4.1/bin/ruby_executable_hooks:15:in `eval'
	from /Users/camm/.rvm/gems/ruby-2.4.1/bin/ruby_executable_hooks:15:in `<main>'

Also, holy project necro!

**Comment by cmaxwellau** [comment]
I've patched this locally, will send a pull request soon:

in intrinsic_function.rb:


```
  def fn_split
    delimiter = '"' + @parameters[0].gsub('"'){'\\"'} + '"'
    # value     = '"' + @parameters[1] + '"'
    if @parameters[1].instance_of? String
       value     = '"' + @parameters[1] + '"'
    else
      true_value = @parameters[1].ai
    end
    return "FnSplit(#{delimiter}, #{value})"
  end
```

**Comment by cmaxwellau** [comment]
Fix is in #19
